fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersMap.java

```diff
@@ -77,12 +101,24 @@ public void insertAll(ParametersMap map) {
         return codeMap.get(searchParameterCode);
     }
 
+    /**
+     * @deprecated use {@link #lookupByCanonical(String)}, which takes an optional version, instead
+     */
+    @Deprecated
     public SearchParameter lookupByUrl(String searchParameterUrl) {
-        return urlMap.get(searchParameterUrl);
+        return canonicalMap.get(searchParameterUrl);
+    }
+
+    public SearchParameter lookupByCanonical(String searchParameterCanonical) {
+        return canonicalMap.get(searchParameterCanonical);
     }
 
     public Collection<SearchParameter> values() {
-        return urlMap.values();
+        // use List to preserve order
+        return Collections.unmodifiableList(canonicalMap.entrySet().stream()
+                .filter(e -> !e.getKey().contains("|"))
+                .map(e -> e.getValue())
+                .collect(Collectors.toList()));
     }
```

Comment:
I think this is a good change.  How expensive is this?

Reply:
good question.  currently its only used from `SearchUtil.getUserDefinedSearchParameters(resourceType)` which is just for the extension parameters.  typically we don't have many of those, but it would be more costly if someone has a lot of them.

Reply:
Great